media-gfx/zbar/zbar-0.23.ebuild

```diff
@@ -0,0 +1,221 @@
+REQUIRED_USE="
+	introspection? ( gtk )
+	python? ( ${PYTHON_REQUIRED_USE} )
+	test? (
+		${PYTHON_REQUIRED_USE}
+		X || ( graphicsmagick imagemagick )
+	)
+	xv? ( X )"
+
+RESTRICT="!test? ( test )"
+
+COMMON_DEPEND="
+	dbus? ( sys-apps/dbus[${MULTILIB_USEDEP}] )
+	gtk? (
+		dev-libs/glib:2[${MULTILIB_USEDEP}]
+		introspection? ( dev-libs/gobject-introspection )
+		x11-libs/gtk+:3[${MULTILIB_USEDEP}]
+	)
+	imagemagick? (
+		!graphicsmagick? (
+			jpeg? ( media-gfx/imagemagick:=[jpeg] )
+			media-gfx/imagemagick:=[png]
+		)
+		graphicsmagick? (
+			jpeg? ( media-gfx/graphicsmagick:=[jpeg] )
+			media-gfx/graphicsmagick:=[png]
+		)
+	)
+	jpeg? ( virtual/jpeg:0[${MULTILIB_USEDEP}] )
+	python? ( ${PYTHON_DEPS} )
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtgui:5
+		dev-qt/qtwidgets:5
+		dev-qt/qtx11extras:5
+	)
+	v4l? ( media-libs/libv4l:0=[${MULTILIB_USEDEP}] )
+	X? (
+		x11-libs/libX11[${MULTILIB_USEDEP}]
+		x11-libs/libXext[${MULTILIB_USEDEP}]
+		xv? ( x11-libs/libXv[${MULTILIB_USEDEP}] )
+	)"
+
+RDEPEND="${COMMON_DEPEND}
+	java? ( virtual/jre:= )"
+
+DEPEND="${COMMON_DEPEND}
+	java? (
+		test? (
+			dev-java/hamcrest-core:1.3
+			dev-java/junit:4
+		)
+		>=virtual/jdk-1.8:=
+	)
+	test? (
+		${PYTHON_DEPS}
+		dev-python/pillow[${PYTHON_USEDEP}]
+	)"
+
+BDEPEND="
+	gtk? ( dev-util/glib-utils )
+	sys-devel/gettext
+	virtual/pkgconfig"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-0.10-errors.patch"
+	"${FILESDIR}/${P}_create_correct_pkconfig_file_for_zbar-qt5.patch"
+	"${FILESDIR}/${P}_fix_detection_of_errors_in_the_v4l_read.patch"
+	"${FILESDIR}/${P}_fix_python_detect.patch"
+	"${FILESDIR}/${P}_fix_Qt5X11Extras_detect.patch"
+	"${FILESDIR}/${P}_reset_conversion_descriptor_after_close.patch"
+)
+
+pkg_setup() {
+	if use python || use test; then
+		python-single-r1_pkg_setup
+	fi
+	use java && java-pkg-opt-2_pkg_setup
+}
+
+src_prepare() {
+	default
+
+	if use python || use test; then
+		if use test; then
+			# make tests happy
+			# because one of the test requires loadable py module from the current ${BUILD_DIR}
+			sed -i \
+				-e "s|PYTHONPATH=@abs_top_srcdir@|PYTHONPATH=@builddir@|g" \
+				test/Makefile.am.inc || die
+		fi
+
+		python_fix_shebang \
+			examples/*.py \
+			test/{test_python,barcodetest}.py # test_pygtk.py — py2 only
+	fi
+
+	if use java; then
+		java-pkg-opt-2_src_prepare
+		sed "s|javadir = \$(pkgdatadir)|javadir = /usr/$(get_libdir)/zbar|" \
+			-i java/Makefile.am || die
+	fi
+
+	eautoreconf
+}
+
+multilib_src_configure() {
+	append-cppflags -DNDEBUG
+
+	local myeconfargs=(
+		$(use_with dbus)
+		$(use_with jpeg)
+		$(multilib_native_use_with introspection gir)
+		$(multilib_native_use_with java)
+		$(multilib_native_use_with python python auto)
+		$(use_enable static-libs static)
+		$(use_enable threads pthread)
+		$(use_enable v4l video)
+		$(use_with X x)
+		$(use_with X xshm)
+		$(use_with xv xv)
+	)
+
+	if use gtk; then
+		myeconfargs+=( --with-gtk=gtk3 )
+	else
+		myeconfargs+=( --with-gtk=no ) # default: 'auto'
+	fi
```

Comment:
`$(use_with gtk3 gtk)`

Reply:
did you mean `$(use_with gtk gtk gtk3)` ?


all of changes ... :
```diff
diff --git a/media-gfx/zbar/zbar-0.23.ebuild b/media-gfx/zbar/zbar-0.23.ebuild
index 025fb5e35df..8a67c136d13 100644
--- a/media-gfx/zbar/zbar-0.23.ebuild
+++ b/media-gfx/zbar/zbar-0.23.ebuild
@@ -18,10 +18,11 @@ KEYWORDS="~amd64 ~arm ~x86"
 IUSE="dbus graphicsmagick gtk +imagemagick introspection java jpeg python qt5 static-libs test +threads v4l X xv"
 REQUIRED_USE="
 	introspection? ( gtk )
+	jpeg? ( imagemagick )
 	python? ( ${PYTHON_REQUIRED_USE} )
 	test? (
 		${PYTHON_REQUIRED_USE}
-		X || ( graphicsmagick imagemagick )
+		X? ( imagemagick )
 	)
 	xv? ( X )"
 
@@ -31,18 +32,12 @@ COMMON_DEPEND="
 	dbus? ( sys-apps/dbus[${MULTILIB_USEDEP}] )
 	gtk? (
 		dev-libs/glib:2[${MULTILIB_USEDEP}]
-		introspection? ( dev-libs/gobject-introspection )
 		x11-libs/gtk+:3[${MULTILIB_USEDEP}]
+		introspection? ( dev-libs/gobject-introspection )
 	)
 	imagemagick? (
-		!graphicsmagick? (
-			jpeg? ( media-gfx/imagemagick:=[jpeg] )
-			media-gfx/imagemagick:=[png]
-		)
-		graphicsmagick? (
-			jpeg? ( media-gfx/graphicsmagick:=[jpeg] )
-			media-gfx/graphicsmagick:=[png]
-		)
+		!graphicsmagick? ( media-gfx/imagemagick:=[png,jpeg?] )
+		graphicsmagick? ( media-gfx/graphicsmagick:=[png,jpeg?] )
 	)
 	jpeg? ( virtual/jpeg:0[${MULTILIB_USEDEP}] )
 	python? ( ${PYTHON_DEPS} )
@@ -64,11 +59,11 @@ RDEPEND="${COMMON_DEPEND}
 
 DEPEND="${COMMON_DEPEND}
 	java? (
+		>=virtual/jdk-1.8:=
 		test? (
 			dev-java/hamcrest-core:1.3
 			dev-java/junit:4
 		)
-		>=virtual/jdk-1.8:=
 	)
 	test? (
 		${PYTHON_DEPS}
@@ -76,9 +71,9 @@ DEPEND="${COMMON_DEPEND}
 	)"
 
 BDEPEND="
-	gtk? ( dev-util/glib-utils )
 	sys-devel/gettext
-	virtual/pkgconfig"
+	virtual/pkgconfig
+	gtk? ( dev-util/glib-utils )"
 
 PATCHES=(
 	"${FILESDIR}/${PN}-0.10-errors.patch"
@@ -127,6 +122,7 @@ multilib_src_configure() {
 
 	local myeconfargs=(
 		$(use_with dbus)
+		$(use_with gtk gtk gtk3) # default is gtk2
 		$(use_with jpeg)
 		$(multilib_native_use_with introspection gir)
 		$(multilib_native_use_with java)
@@ -139,12 +135,6 @@ multilib_src_configure() {
 		$(use_with xv xv)
 	)
 
-	if use gtk; then
-		myeconfargs+=( --with-gtk=gtk3 )
-	else
-		myeconfargs+=( --with-gtk=no ) # default: 'auto'
-	fi
-
 	if multilib_is_native_abi; then
 		# both must be enabled to use GraphicsMagick
 		if use graphicsmagick; then
@@ -212,7 +202,6 @@ src_install() {
 }
 
 multilib_src_install_all() {
-	einstalldocs
 	find "${D}" -name '*.la' -delete || die
 }

```